app/src/main/java/org/oppia/android/app/spotlight/SpotlightFragment.kt

```diff
@@ -0,0 +1,448 @@
+    featureViewStateLiveData.observe(
+      activity,
+      object : Observer<AsyncResult<SpotlightViewState>> {
+        override fun onChanged(it: AsyncResult<SpotlightViewState>?) {
+          if (it is AsyncResult.Success) {
+            val viewState = (it.value)
+            if (viewState == SpotlightViewState.SPOTLIGHT_SEEN) {
+              return
+            } else if (viewState == SpotlightViewState.SPOTLIGHT_NOT_SEEN) {
+              createTarget(spotlightTarget)
+              featureViewStateLiveData.removeObserver(this)
+            }
+          }
+        }
+      }
+    )
+  }
+
+  private fun createTarget(
+    spotlightTarget: SpotlightTarget
+  ) {
+
+    val target = Target.Builder()
+      .setShape(getShape(spotlightTarget))
+      .setOverlay(requestOverlayResource(spotlightTarget))
+      .setOnTargetListener(object : OnTargetListener {
+        override fun onStarted() {
+
+          Log.d("overlay", "target  start")
+        }
+
+        override fun onEnded() {
+          Log.d("overlay", "target  end")
+          targetList.pop()
+//          val profileId = ProfileId.newBuilder()
+//            .setInternalId(internalProfileId)
+//            .build()
+//          spotlightStateController.markSpotlightViewed(profileId, spotlightTarget.feature)
+        }
+      })
+      .build(spotlightTarget.anchor)
+
+    targetList.add(target)
+    if (!isSpotlightActive.value!!) {
+      startSpotlight()
+    }
+  }
+
+  private fun startSpotlight() {
+    if (targetList.isNullOrEmpty()) return
+    spotlight = Spotlight.Builder(activity)
+      .setTargets(targetList)
+      .setBackgroundColorRes(R.color.spotlightBackground)
+      .setDuration(1000L)
+      .setAnimation(DecelerateInterpolator(2f))
+      .setOnSpotlightListener(object : OnSpotlightListener {
+        override fun onStarted() {
+          Log.d("overlay", "spotlight  start")
+
+          isSpotlightActive.value = true
+        }
+
+        override fun onEnded() {
+          Log.d("overlay", "spotlight  end")
+
+          isSpotlightActive.value = false
+          startSpotlight()
+        }
+      })
+      .build()
+
+    spotlight.start()
+  }
+
+  private fun getShape(spotlightTarget: SpotlightTarget): Shape {
+    return when (spotlightTarget.shape) {
+      SpotlightShape.RoundedRectangle -> {
+        RoundedRectangle(
+          spotlightTarget.anchorHeight.toFloat(),
+          spotlightTarget.anchorWidth.toFloat(),
+          24f
+        )
+      }
+      SpotlightShape.Circle -> {
+        return if (spotlightTarget.anchorHeight > spotlightTarget.anchorWidth) {
+          Circle((spotlightTarget.anchorHeight / 2).toFloat())
+        } else {
+          Circle((spotlightTarget.anchorWidth / 2).toFloat())
+        }
+      }
+    }
+  }
+
+  private fun checkIsRTL() {
+    val locale = Locale.getDefault()
+    val directionality: Byte = Character.getDirectionality(locale.displayName[0].toInt())
```

Comment:
You can use https://github.com/oppia/oppia-android/blob/5e64fae55e0024529ac7eb48de401c2035d2e4f3/utility/src/main/java/org/oppia/android/util/locale/OppiaLocale.kt#L263 for this purpose, which is available to app module components via: https://github.com/oppia/oppia-android/blob/5e64fae55e0024529ac7eb48de401c2035d2e4f3/app/src/main/java/org/oppia/android/app/translation/AppLanguageResourceHandler.kt#L139

Reply:
This has been utilized & addressed in the latest commits.